frontend/javascripts/dashboard/dataset/queries.tsx

```diff
@@ -498,3 +518,58 @@ function getUnobtrusivelyUpdatedDatasets(
     };
   });
 }
+
+type FolderHierarchy = {
+  tree: FolderItem[];
+  itemById: Record<string, FolderItem>;
+  flatItems: FlatFolderTreeItem[];
+};
+
+export function getFolderHierarchy(folderTree: FlatFolderTreeItem[]): FolderHierarchy {
```

Comment:
this function was basically moved from `frontend/javascripts/dashboard/folders/folder_tree.tsx`.